src/local/butler/reproduce_tool/http_utils.py

```diff
@@ -0,0 +1,101 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Common utility functions for the reproduce tool."""
+
+import httplib2
+import os
+import webbrowser
+
+from base import json_utils
+from base import utils
+from system import shell
+
+GET_METHOD = 'GET'
+POST_METHOD = 'POST'
+
+
+class SuppressOutput(object):
+  """Suppress stdout and stderr.
+
+  We need this to suppress webbrowser's stdout and stderr."""
+
+  def __enter__(self):
+    self.stdout = os.dup(1)
+    self.stderr = os.dup(2)
+    os.close(1)
+    os.close(2)
+    os.open(os.devnull, os.O_RDWR)
+
+  def __exit__(self, *_):
+    os.dup2(self.stdout, 1)
+    os.dup2(self.stderr, 2)
+    return True
+
+
+def _get_authorization(force_reauthorization):
+  """Get the value for an oauth authorization header."""
+  # Avoid a circular import.
+  from local.butler.reproduce_tool import config
```

Comment:
Since this code is all new, we need to figure out the circular import issue. this is not nice in so many functions.

Reply:
Done